libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -544,6 +545,21 @@ - (void) syncUpOneEntry:(SFSyncState*)sync recordIds:(NSArray*)recordIds index:(
     NSString* objectType = [SFJsonUtils projectIntoJson:record path:kSyncManagerObjectTypePath];
     NSString* objectId = record[kSyncManagerObjectId];
     NSNumber* soupEntryId = record[SOUP_ENTRY_ID];
+    NSString* lastModifiedDateString = record[kSyncManagerLastModifiedDate];
+    long long lastModifiedDate = [self getTimeMillisFromString:lastModifiedDateString];
+
+    /*
+     * Checks if we are attempting to update a record that has been updated
+     * on the server AFTER the client's last sync down. If the merge mode
+     * passed in tells us to leave the record alone under these
+     * circumstances, we will do nothing and return here.
+     */
+    if (mergeMode == SFSyncStateMergeModeLeaveIfChanged &&
+        (action == kSyncManagerActionUpdate || action == kSyncManagerActionDelete) &&
+        ![self isNewerThanServer:objectType objectId:objectId lastModifiedDate:lastModifiedDate]) {
+        // Next
+        [self syncUpOneEntry:sync recordIds:recordIds index:i+1 updateSync:updateSync failRest:failRest];
+    }
```

Comment:
Good catch! I was blindly following some previously written code a few lines above for the same use case. I'll add return statements to both.